pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -554,23 +495,24 @@ func (c *Controller) syncCheckStates(ctx context.Context, wl *kueue.Workload, ch
 			}
 		}
 
+		wlPatch := workload.BaseSSAWorkload(wl)
 		existingCondition := workload.FindAdmissionCheck(wlPatch.Status.AdmissionChecks, checkState.Name)
+		var message string
 		if existingCondition != nil && existingCondition.State != checkState.State {
-			message := fmt.Sprintf("Admission check %s updated state from %s to %s", checkState.Name, existingCondition.State, checkState.State)
+			message = fmt.Sprintf("Admission check %s updated state from %s to %s", checkState.Name, existingCondition.State, checkState.State)
 			if checkState.Message != "" {
 				message += fmt.Sprintf(" with message %s", checkState.Message)
 			}
-			recorderMessages = append(recorderMessages, message)
 		}
-
 		workload.SetAdmissionCheckState(&wlPatch.Status.AdmissionChecks, checkState)
-	}
-	if updated {
-		if err := c.client.Status().Patch(ctx, wlPatch, client.Apply, client.FieldOwner(ControllerName), client.ForceOwnership); err != nil {
-			return err
-		}
-		for i := range recorderMessages {
-			c.record.Event(wl, corev1.EventTypeNormal, "AdmissionCheckUpdated", api.TruncateEventMessage(recorderMessages[i]))
+
+		if updated {
+			if err := c.client.Status().Patch(ctx, wlPatch, client.Apply, client.FieldOwner(ControllerName), client.ForceOwnership); err != nil {
```

Comment:
Any reason to send the workload updates per admission checks?

We used to send one build Patch for all updates to the workload which is more performant. In practice there should not be too many ACs, but still it is preferred to combine the workload updates, if possible.

Reply:
Fixed, thanks